### PR TITLE
Add new OpenStack module for managing cinder volume snapshots

### PR DESCRIPTION
**Pull request by ruizink**
##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/openstack/os_volume_snapshot.py
##### ANSIBLE VERSION

```
ansible 2.6.0.0
```
##### SUMMARY

This change adds a new OpenStack module to allow the creation/deletion of snapshots on cinder volumes.

### CODE REVIEW

**Comment by ruizink** [comment]
Any updates regarding this MR?

File: lib/ansible/modules/cloud/openstack/os_volume_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
```
Comment (maxamillion) [inline lib/ansible/modules/cloud/openstack/os_volume_snapshot.py]: Needs an `ANSIBLE_METADATA` entry as per https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html

File: lib/ansible/modules/cloud/openstack/os_volume_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+DOCUMENTATION = '''
+---
+module: os_volume_snapshot
+short_description: Create/Delete Cinder Volume Snapshots
+extends_documentation_fragment: openstack
+version_added: "2.3"
```
Comment (maxamillion) [inline lib/ansible/modules/cloud/openstack/os_volume_snapshot.py]: Just a note, we'll need to bump this to the correct version at the time this gets included.

**Review by maxamillion — CHANGES_REQUESTED** [review]

**Comment by maxamillion** [comment]
As a general note, any desire to support `check_mode`?

**Comment by maxamillion** [comment]
needs_info needs_revision

**Comment by ruizink** [comment]
* Rebased from `devel`
* Bumped the Ansible version in `version_added`
* Added `ANSIBLE_METADATA` section
* Added support for `check_mode`

**Comment by Akasurde** [comment]
@maxamillion Could you please revisit changes ?

**Review by maxamillion — APPROVED** [review]

**Comment by maxamillion** [comment]
I'd like to note that I should have encouraged adding integration tests previously, but I won't block on it. The code looks good.

**Comment by ruizink** [comment]
@maxamillion I will work on adding those. Thanks

**Comment by maxamillion** [comment]
@ruizink much appreciated!

**Pull request merged**